### PR TITLE
ccPropertiesTreeDelegate null pointer checks

### LINKED ISSUE #1282: ccPropertiesTreeDelegate unguarded casts lead to crash
crash occurred when mouse wheel scrolling through the properties window of a plane object. for some reason the 
image
image
I believe this can occur when rapidly changing selected menu items and having a que of actions being executed after the base item has been changed. in this case the cast to sphere fails because the m_currentObject was equal to the plane object not the sphere it was previously.

I modified ccPropertiesTreeDelegate.cpp to add guards at each cast, I'll submit a PR later tonight.

### PR DESCRIPTION
**Pull request by WargodHernandez**
Adds guards against null pointer dereferences in ccPropertiesTreeDelegate
fixes #1282

Also adds braces around single line if statements in this file

### CODE REVIEW

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 1]
```diff
@@ -400,26 +413,35 @@ void ccPropertiesTreeDelegate::fillWithMetaData(const ccObject* _obj)
 void ccPropertiesTreeDelegate::fillWithHObject(ccHObject* _obj)
 {
 	assert(_obj && m_model);
-
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:416]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 2]
```diff
@@ -470,12 +492,18 @@ void ccPropertiesTreeDelegate::fillWithHObject(ccHObject* _obj)
...
-		appendRow(ITEM( tr( "Current Display" )), PERSISTENT_EDITOR(OBJECT_CURRENT_DISPLAY), true);
+	{
+		appendRow(ITEM(tr("Current Display")), PERSISTENT_EDITOR(OBJECT_CURRENT_DISPLAY), true);
+	}
 }
 
 void ccPropertiesTreeDelegate::fillWithShifted(const ccShiftedObject* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:503]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 3]
```diff
@@ -488,7 +516,10 @@ void ccPropertiesTreeDelegate::fillWithShifted(const ccShiftedObject* _obj)
 void ccPropertiesTreeDelegate::fillWithPointCloud(ccGenericPointCloud* _obj)
 {
 	assert(_obj && m_model);
-
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:519]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 4]
```diff
@@ -541,11 +576,16 @@ void ccPropertiesTreeDelegate::fillWithPointCloud(ccGenericPointCloud* _obj)
 void ccPropertiesTreeDelegate::fillSFWithPointCloud(ccGenericPointCloud* _obj)
 {
 	assert(m_model);
-
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:579]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 5]
```diff
@@ -585,6 +625,10 @@ void ccPropertiesTreeDelegate::fillSFWithPointCloud(ccGenericPointCloud* _obj)
 void ccPropertiesTreeDelegate::fillWithPrimitive(const ccGenericPrimitive* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:628]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 6]
```diff
@@ -624,6 +668,10 @@ void ccPropertiesTreeDelegate::fillWithPrimitive(const ccGenericPrimitive* _obj)
 void ccPropertiesTreeDelegate::fillWithFacet(const ccFacet* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:671]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 7]
```diff
@@ -650,6 +698,10 @@ void ccPropertiesTreeDelegate::fillWithFacet(const ccFacet* _obj)
 
 void ccPropertiesTreeDelegate::fillWithPlanarEntity(const ccPlanarEntityInterface* _obj)
 {
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:701]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 8]
```diff
@@ -668,6 +720,10 @@ void ccPropertiesTreeDelegate::fillWithPlanarEntity(const ccPlanarEntityInterfac
 void ccPropertiesTreeDelegate::fillWithMesh(const ccGenericMesh* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:723]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 9]
```diff
@@ -696,6 +752,10 @@ void ccPropertiesTreeDelegate::fillWithMesh(const ccGenericMesh* _obj)
 void ccPropertiesTreeDelegate::fillWithPolyline(const ccPolyline* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:755]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 10]
```diff
@@ -715,6 +775,10 @@ void ccPropertiesTreeDelegate::fillWithPolyline(const ccPolyline* _obj)
 void ccPropertiesTreeDelegate::fillWithPointOctree(const ccOctree* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:778]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 11]
```diff
@@ -745,6 +809,10 @@ void ccPropertiesTreeDelegate::fillWithPointOctree(const ccOctree* _obj)
 void ccPropertiesTreeDelegate::fillWithPointKdTree(const ccKdTree* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:812]: || ?

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp [hunk 12]
```diff
@@ -782,6 +850,10 @@ void ccPropertiesTreeDelegate::fillWithPointKdTree(const ccKdTree* _obj)
 void ccPropertiesTreeDelegate::fillWithImage(const ccImage* _obj)
 {
 	assert(_obj && m_model);
+	if (!_obj && !m_model)
```
Comment (dgirardeau) [inline qCC/db_tree/ccPropertiesTreeDelegate.cpp:853]: Okay I'll stop now :D

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp (hunk 1 above)
Comment (WargodHernandez) [reply qCC/db_tree/ccPropertiesTreeDelegate.cpp:416]: Darn copy and paste! :) I'll get these switched over

File: qCC/db_tree/ccPropertiesTreeDelegate.cpp (hunk 12 above)
Comment (WargodHernandez) [reply qCC/db_tree/ccPropertiesTreeDelegate.cpp]: :)

**Pull request merged**